### PR TITLE
Fix InfluxDB unhealthy alerts and false "no data"s

### PR DESCRIPTION
**Pull request by santihernandezc**
**What this PR does / why we need it**:

When you create an InfluxDB alert and try to run it, it's very possible that one of the frames returned by the query is of type `TimeSeriesTypeNot`. This causes the function `expr.WideToMany()` to error out. The result is an API error and a "No Data" message.

You can run the exact same query in the Explore tab and get results. It's a false "No Data" response.

When you configure the alert to fire when there's no data returned and save it, the alert immediately fires due to a false "No Data" response. Also, it shows `error` under the "Health" column.

Screen Shot 2021-12-21 at 15 25 31

To work around inconsistencies between data source responses, a check has been added to act differently when it comes to InfluxDB queries. The query is performed correctly and the alert is healthy.

Screen Shot 2021-12-21 at 15 30 35

alerta no data healthy

This is a temporary fix to solve an issue some users are facing. A more definitive solution should be implemented.
**Which issue(s) this PR fixes**:

https://github.com/grafana/support-escalations/issues/1281

### CODE REVIEW

**Comment by davidmparrott** [comment]
Is there a case where `TimeSeriesTypeNot` is a valid response from InfluxDB that this will short circuit?

**Comment by gerobinson** [comment]
Do we have a better understanding of what `TimeSeriesTypeNot` means for InfluxDB? I think this is a good change but like David said I think we need to understand it a little better before approving this.

File: pkg/expr/nodes.go [hunk 1]
```diff
@@ -254,8 +254,14 @@ func (dn *DSNode) Execute(ctx context.Context, vars mathexp.Vars, s *Service) (m
 			}
 		}
 
+		dataSource := dn.datasource.Type
 		for _, frame := range qr.Frames {
 			logger.Debug("expression datasource query (seriesSet)", "query", refID)
+			// Check for TimeSeriesTypeNot in an InfluxDB query.
+			// This check should be removed once inconsistencies in data source responses are solved.
+			if frame.TimeSeriesSchema().Type == data.TimeSeriesTypeNot && dataSource == models.DS_INFLUXDB {
+				continue
```
Comment (kylebrandt) [inline pkg/expr/nodes.go:263]: Do we know anything more specific about the shape of the frame (the fields) that we want to drop that we could could check for and/or comment on?

Comment (kylebrandt) [reply pkg/expr/nodes.go:263]: Also perhaps log a warning here? Seems like we don't want to silently drop, that way if there are future issues with influx, that logging information can be used to see if this is being hit.

**Comment by santihernandezc** [comment]
@davidmparrott @gerobinson I can't think of any case where this piece of code would short circuit or affect the alerts.

The `WideToMany()` function converts a data frame into a slice of `mathexp.Series`, which must have `time.Time` and a `*float64` fields. The data frames of type `TimeSeriesTypeNot` don't have any numeric fields.

For example, a data frame of this type could have `*time.Time` and `*[]string` fields, thus not being relevant for evaluating an alert condition. This PR prevents these "irrelevant" data frames from being evaluated and resulting in unexpected errors and unhealthy alerts. Examples of this kind of data frames in an InfluxDB response are shown below.

Screen Shot 2022-01-04 at 15 01 26

Screen Shot 2022-01-04 at 15 02 35

**Comment by gerobinson** [comment]
Thanks @santihernandezc! I somehow forgot that InfluxDB is not restricted to just `float64` but can also represent other data such as strings. Thanks!

**Review by yuri-tceretian — APPROVED** [review]
LGTM as a temp fix. However, I would like to better understand what could be the proper solution. 
Also, it is not clear what is the impact on explore\panel as they use the same mechanism to query data. Are there any visible changes in panel and\or explore?

**Comment by santihernandezc** [comment]
@yuri-tceretian 
Both the Explore tab and the alert panel use the same mechanism to query the data (`*influxdb.Service.QueryData()`), but the Explore tab's handler returns a `*backend.QueryDataResponse` in JSON format, as it's used for visualization purposes only, while the alert panel uses the function `WideToMany()` as part of the alert evaluation pipeline. That's where lies the difference.

The alert page's handler ranges through that `*backend.QueryDataResponse`, applying `WideToMany()` to each data frame to get a `[]mathexp.Series`. This is where unexpected errors occur as a result of non-numeric data frames.

I did not notice any differences between the Explore and alert panels besides the expected visual ones.

Screen Shot 2022-01-26 at 17 48 19

Screen Shot 2022-01-26 at 17 48 26

**Pull request merged**